File: api/src/main/java/com/ruby/rubia_server/core/entity/Campaign.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.base.BaseEntity;
import com.ruby.rubia_server.core.enums.CampaignStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.type.SqlTypes;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "campaigns")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Campaign implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "campaign_status", nullable = false)
    @Builder.Default
    private CampaignStatus status = CampaignStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdBy;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "target_audience_description", columnDefinition = "TEXT")
    private String targetAudienceDescription;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "message_template_id")
    private MessageTemplate initialMessageTemplate;

    @Column(name = "total_contacts")
    @Builder.Default
    private Integer totalContacts = 0;

    @Column(name = "contacts_reached")
    @Builder.Default
    private Integer contactsReached = 0;

    @Column(name = "source_system_name")
    private String sourceSystemName; // Ex: "CRM Externo", "Sistema de Doação Realblood"

    @Column(name = "source_system_id")
    private String sourceSystemId; // ID original da campanha no sistema de origem

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/Campaign.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("campaigns")]
public class Campaign : BaseEntity
{
    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("description", TypeName = "TEXT")]
    public string? Description { get; set; }

    [Column("campaign_status")]
    [Required]
    public CampaignStatus Status { get; set; } = CampaignStatus.Active;

    [Column("start_date")]
    public DateOnly? StartDate { get; set; }

    [Column("end_date")]
    public DateOnly? EndDate { get; set; }

    [Column("target_audience_description", TypeName = "TEXT")]
    public string? TargetAudienceDescription { get; set; }

    [Column("total_contacts")]
    public int TotalContacts { get; set; } = 0;

    [Column("contacts_reached")]
    public int ContactsReached { get; set; } = 0;

    [Column("source_system_name")]
    public string? SourceSystemName { get; set; } // Ex: "CRM Externo", "Sistema de Doação Realblood"

    [Column("source_system_id")]
    public string? SourceSystemId { get; set; } // ID original da campanha no sistema de origem

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    [Column("created_by_user_id")]
    public Guid? CreatedByUserId { get; set; }
    
    [ForeignKey("CreatedByUserId")]
    public virtual User? CreatedBy { get; set; }

    [Column("message_template_id")]
    public Guid? MessageTemplateId { get; set; }
    
    [ForeignKey("MessageTemplateId")]
    public virtual MessageTemplate? InitialMessageTemplate { get; set; }

    public virtual ICollection<CampaignContact> CampaignContacts { get; set; } = new List<CampaignContact>();
    public virtual ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();
}